library/std/src/path.rs

```diff
@@ -3667,11 +3679,36 @@ impl Error for NormalizeError {}
 /// # fn main() {}
 /// ```
 ///
+/// ## Cygwin paths
+///
+/// ```
+/// # #[cfg(target_os = "cygwin")]
+/// fn main() -> std::io::Result<()> {
+///     use std::path::{self, Path};
+///
+///     // Relative to absolute
+///     let absolute = path::absolute("foo/./bar")?;
+///     assert!(absolute.ends_with(r"foo/bar"));
+///
+///     // Windows absolute to absolute
+///     let absolute = path::absolute(r"C:\foo//test\..\./bar.rs")?;
+///     assert!(absolute.ends_with("/c/foo/bar.rs"));
+///
+///     // POSIX absolute to absolute
+///     let absolute = path::absolute("/foo//test/.././bar.rs")?;
+///     assert_eq!(absolute, Path::new("/foo//test/.././bar.rs"));
```

Comment:
Hm, I would expect the `./` to be removed to match the unix behaviour.

Reply:
Currently `absolute` is backed by `cygwin_conv_path` which handles all kinds of paths, both POSIX and Windows one. It's the only solution because only cygwin knows where Windows drives are mounted. Or, should I process the returned path of it, with the code in `unix.rs`?

Reply:
OK, updated the code and comments.

Reply:
I am quite surprised that `cygwin_conv_path` didn't normalize the path.  Maybe because it was already absolute?
```console
$ cd /cygdrive/c
$ cygpath -ua /foo//test/.././bar.rs
/foo//test/.././bar.rs
$ cygpath -ua foo//test/.././bar.rs
/cygdrive/c/foo/bar.rs
```

Reply:
I think so. Well it's OK because we can normalize ourselves.